### PR TITLE
Update TC-DD-3.2/3.6/3.8 Non-Concurrent(#31751)

### LINKED ISSUE #31751: [1.3] Update TH YAML scripts TC-DD-3.2/3.6/3.8 to meet new spec
### Reproduction steps

[TC-DD-3.2] [TC-DD-3.6] and [TC-DD-3.8]

The test plan and SDK for the above Non-concurrent tests has been updated and need to reflected in the TH YAML.

See Test Plan pull request https://github.com/CHIP-Specifications/chip-test-plans/pull/3893.
See SDK pull request https://github.com/project-chip/connectedhomeip/pull/31739

This Test Plan change and the corresponding SDK change need approving and merging before this issue and corresponding PR can be merged.

### Bug prevalence

Every time on commissionee when running in Non-concurrent mode

### GitHub hash of the SDK that was being used

f63b505179

### Platform

raspi

### Platform Version(s)

_No response_

### Type

Spec Compliance Issue

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by simonhmorris1**
Fixes #31751 Update YAML TH TC-DD3.2/3.6/3.8 to meet latest Matter spec.
This **should be merged after SDK change #31739** has been merged.

This updates the Manual test steps to take into account Non-concurrent mode.

This requires the all-clusters test harness to be built with Non-concurrent mode disabled.
That can be achieved by compiling with the following change. 

**Is this acceptable or does the all-clusters app need to be able to select this at runtime on the command line?**

```
examples/all-clusters-app/linux/include/CHIPProjectAppConfig.h
#define CHIP_DEVICE_CONFIG_SUPPORTS_CONCURRENT_CONNECTION 0
```

```
cd connectedhomeip
./scripts/run_in_build_env.sh "./scripts/build/build_examples.py --target linux-arm64-all-clusters-asan build"
```

### CODE REVIEW

**Review by swan-amazon — APPROVED** [review]

**Review by liamgonyea — APPROVED** [review]
Changes here corresponding to https://github.com/CHIP-Specifications/chip-test-plans/pull/3893 look good.

File: src/app/tests/suites/certification/Test_TC_DD_3_8.yaml [hunk 1]
```diff
@@ -64,6 +66,30 @@ tests:
           CHIP:SC: Received PBKDF param response
       disabled: true
 
+    - label:
+          "Step 3a: Commissioner reads commissioning information, including
+          SupportsConcurrentConnection attribute from General Commissioining
```
Comment (Damian-Nordic) [inline src/app/tests/suites/certification/Test_TC_DD_3_8.yaml:71]:
```suggestion
          SupportsConcurrentConnection attribute from General Commissioning
```

File: src/app/tests/suites/certification/Test_TC_DD_3_8.yaml [hunk 2]
```diff
@@ -64,6 +66,30 @@ tests:
...
+          Cluster"
+      verification: |
+          Verify in TH that SupportsConcurrentConnection is false
+
+          CHIP:IM: Received Read request
```
Comment (Damian-Nordic) [inline src/app/tests/suites/certification/Test_TC_DD_3_8.yaml:76]: How do the log tell that SupportsConcurrentConnection is false?

Comment (Damian-Nordic) [reply src/app/tests/suites/certification/Test_TC_DD_3_8.yaml:76]: nvm, I only noticed it tests the commissioner.

**Review by Damian-Nordic — APPROVED** [review]

**Review by chrisdecenzo — APPROVED** [review]

**Pull request merged**